transaction/core/src/mint/validation/config.rs

```diff
@@ -130,20 +146,44 @@ mod tests {
         };
         let mint_config4 = MintConfig {
             token_id: *token_id,
-            signer_set: SignerSet::new(vec![signer_2.public_key(), signer_3.public_key()], 2),
+            signer_set: SignerSet::new_with_multi(
+                vec![signer_1.public_key()],
+                vec![SignerSet::new(
+                    vec![signer_2.public_key(), signer_3.public_key()],
+                    2,
+                )],
+                2,
+            ),
             mint_limit: 15,
         };
 
-        assert!(validate_configs(
-            token_id,
-            &[mint_config1, mint_config2, mint_config3, mint_config4]
-        )
-        .is_ok());
+        for block_version in BlockVersion::iterator() {
+            if !block_version.mint_transactions_are_supported() {
+                continue;
+            }
+
+            let mut configs = vec![
+                mint_config1.clone(),
+                mint_config2.clone(),
+                mint_config3.clone(),
+            ];
+            if block_version.nested_multisigs_are_supported() {
+                configs.push(mint_config4.clone());
+            }
+
+            assert!(validate_configs(token_id, &configs, block_version,).is_ok());
+        }
     }
 
     #[test]
     fn validate_configs_accepts_no_configs() {
-        assert!(validate_configs(123.into(), &[]).is_ok());
+        for block_version in BlockVersion::iterator() {
+            if !block_version.mint_transactions_are_supported() {
+                continue;
+            }
```

Comment:
I don't agree with that, since if we end up moving the block version at which we release this feature, we're gonna need to remember to go and bump the number in all these tests...
I don't think we care too much about the specific block version number, we just want to test all versions that happen to support mint transactions. For example, it is possible that at some point in the future they will get deprecated and we will stop supporting them and for some period the code will continue to exist (especially during the transition period). If we use straight up block version numbers we'll then need to also limit to the upper bound, and again hardcode a number in a bunch of tests. I think it's fairly clear to the reader/future maintainer what is happening here.

Reply:
(with that said, I don't feel strongly about this so if `.skip(2)` gets this to move quicker down the review pipeline I am happy to switch to that)

Reply:
I can buy the argument of avoiding hardcoding block versions to minimize test maintenance. (This and other comments)

Reply:
If that feels good to you then I will mark the comments related to this as resolved.